### PR TITLE
Fix stack push/pop count when encountering a method with a modreq return type

### PR DESCRIPTION
**Pull request by zsr2531**
This PR fixes an issue with the `GetStackPopCount` and `GetStackPushCount` methods where if the return type is a `CustomModifierTypeSignature`, whose `BaseType` is `void`, it would get treated as if it actually returned a value.

### CODE REVIEW

File: src/AsmResolver.DotNet/Code/Cil/CilInstructionExtensions.cs [hunk 1]
```diff
@@ -23,8 +24,8 @@ public static class CilInstructionExtensions
         public static int GetStackPopCount(this CilInstruction instruction, CilMethodBody? parent)
         {
             return GetStackPopCount(instruction,
-                parent is null
-                || !(parent.Owner.Signature?.ReturnsValue).GetValueOrDefault());
+                parent == null
+                || HasReturnValue(parent.Owner.Signature));
```
Comment (Washi1337) [inline src/AsmResolver.DotNet/Code/Cil/CilInstructionExtensions.cs:28]: Possible NRE when the `Signature` property is `null` (as a result of invalid metadata).

File: src/AsmResolver.DotNet/Code/Cil/CilInstructionExtensions.cs [hunk 2]
```diff
@@ -242,5 +243,13 @@ public static Parameter GetParameter(this CilInstruction instruction, ParameterC
             }
         }
 
+        private static bool HasReturnValue(MethodSignature signature)
+        {
+            var ret = signature.ReturnType;
+            while (ret is CustomModifierTypeSignature custom)
+                ret = custom.BaseType;
+
+            return ret?.IsTypeOf("System", "Void") ?? false;
```
Comment (Washi1337) [inline src/AsmResolver.DotNet/Code/Cil/CilInstructionExtensions.cs:252]: Unnecessary null-coalescing.

**Review by Washi1337 — CHANGES_REQUESTED** [review]

**Comment by Washi1337** [comment]
Actually, I just realized that `MethodSignatureBase` already defines a property `ReturnsValue`. We should probably move the impl of `HasReturnValue` to that property instead.

**Comment by zsr2531** [comment]
Closing because I'll open another PR that is based on the correct branch.

**Pull request closed without merging**